packages/module-source/src-xs/index.js

```diff
@@ -0,0 +1,59 @@
+// @ts-check
+/* global globalThis */
+/* eslint-disable @endo/no-nullish-coalescing */
+
+/** @typedef {{ import: string, as?: string, from: string } & { importAllFrom: string, as: string, from: string } & { export: string, as?: string, from?: string } & { exportAllFrom: string, as?: string } & {importFrom: string }} Binding */
+
+/** @param {Binding[]} bindings */
+function* getImports(bindings) {
+  for (const binding of bindings) {
+    if (binding.import !== undefined) {
+      yield binding.from;
+    } else if (binding.importFrom !== undefined) {
+      yield binding.importFrom;
+    } else if (binding.importAllFrom !== undefined) {
+      yield binding.importAllFrom;
+    } else if (binding.exportAllFrom !== undefined) {
+      yield binding.exportAllFrom;
+    }
+  }
+}
+
+/** @param {Binding[]} bindings */
+function* getExports(bindings) {
+  for (const binding of bindings) {
+    if (binding.export !== undefined) {
+      yield binding.as ?? binding.export;
+    }
+  }
+}
+
+/** @param {Binding[]} bindings */
+function* getReexports(bindings) {
+  for (const binding of bindings) {
+    if (binding.exportAllFrom !== undefined) {
+      yield binding.exportAllFrom;
+    }
+  }
+}
+
+const ModuleSource = globalThis.ModuleSource;
+
+Object.defineProperties(
+  ModuleSource.prototype,
+  Object.getOwnPropertyDescriptors({
+    get imports() {
```

Comment:
I like this approach for adding accessors to an existing object. No change suggested.